lib/verifier.rb

```diff
@@ -17,17 +17,23 @@
 class Verifier
 
   class Result
-    attr_accessor :errors, :name
+    attr_accessor :errors, :warnings, :name
     
     def initialize
       @valid = false
+      @safe = false
       @errors = Array.new
+      @warnings = Array.new
     end
     
     def valid?
       @errors.empty?
     end
 
+    def safe?
```

Comment:
`safe?` is not a very specific name. Maybe better use something like `has_warnings?`.